tests/src/JIT/HardwareIntrinsics/Add.cs

```diff
@@ -0,0 +1,403 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics;
+
+namespace IntelHardwareIntrinsicTest
+{
+    class Program
+    {
+        const int Pass = 100;
+        const int Fail = 0;
+
+        static unsafe int Main(string[] args)
+        {
+            int testResult = Pass;
+
+            if (Avx.IsSupported)
+            {
+                TestTable<float> floatTable = new TestTable<float>(new float[8] {1, -5, 100, 0, 1, -5, 100, 0}, new float[8] {22, -1, -50, 0, 22, -1, -50, 0}, new float[8]);
+                TestTable<double> doubleTable = new TestTable<double>(new double[4] {1, -5, 100, 0}, new double[4] {22, -1, -50, 0}, new double[4]);
+                var vf1 = Unsafe.Read<Vector256<float>>(Unsafe.AsPointer(ref floatTable.inArray1[0]));
```

Comment:
Does these inner arrays need to be `fixed` or `GCHandle.Alloc(floatTable.inArray1,GCHandleType.Pinned)` before grabbing a pointer?

Could add it to `TestTable` and unpin in `Dispose`; so wrap in a using
```csharp
using (TestTable<float> floatTable = new TestTable<float>(....))
using (TestTable<double> floatTable = new TestTable<double>(....))
{
    var vf1 = Unsafe.Read<Vector256<float>>(Unsafe.AsPointer(ref floatTable.inArray1[0]));
    // ...
}
```